### PR TITLE
Report URLs of pending status checks

### LINKED ISSUE #26: Display the current build status from CI in the Hoff dashboard
It would be nice to have a link to the build on the `testing` branch.
On GH you only see the reference to Semaphore on the branch of that PR, but that is the status before the branch was rebased on `testing`.
This is slightly confusing, if you are not aware of that.

We could add this link in the Hoff dashboard , then one could easily access it.

### LINKED ISSUE #109: Make Hoff post the URL to the CI run when merging
Currently when merging a PR Hoff will comment "Rebased as `abcd123`, waiting for CI …" once it has rebased the branch to be merged.
While this is happening the original branch will be untouched, if the tests pass on that branch it will seem as if the tests for the branch to be merged are done already. While this is happening Hoff is actually waiting for a different CI run to complete, which can currently only be reached by manually navigating to the rebased commit's tests.

It would be a better experience if Hoff would post the CI link to the PR that is being merged once this link is known.

Todo:
1. Wait for webhook confirming that a CI run has started
2. Post comment on PR that CI has started with this link

### PR DESCRIPTION
**Pull request by alex-mckenna**
This PR allows `hoff` to report the `target_url` of pending builds, giving users an indication of which job `hoff` is waiting for. This is shown both as a comment on the PR, and an additional link in the headline for pull requests where a CI URL is available (i.e. pending and failed jobs, since we don't currently keep the URL of succeeded jobs).

Screenshots:
Screenshot from 2022-07-21 15-15-17

Screenshot from 2022-07-21 15-15-07

I'm slightly unsure if this will actually go off in practice. This info is given when the build status changes to pending with a non-null `target_url`, but I have no concrete intuition for when this would occur in practice. I'm hoping when the status check starts it will change the state from `BuildPending Nothing` to `BuildPending (Just url)`.

Closes #26
Closes #109

### CODE REVIEW

File: src/Logic.hs
Comment (rudymatela) [inline src/Logic.hs:550]: itym user:
```suggestion
            -- Sometimes we want to leave an informative comment on the PR. This isn't for things
            -- that require user feedback like build failures.
```

File: src/WebInterface.hs
Comment (rudymatela) [inline src/WebInterface.hs:248]: Awesome... and pretty handy!

File: tools/send-webhook
Comment (rudymatela) [inline tools/send-webhook:127]: Code is misaligned here due to a mix of tabs and spaces.  Changing the 8 spaces before target to a tab fixes it:

```suggestion
	event=
	commit="$1"
	state="$2"
	target="$3"
```

File: tools/send-webhook
Comment (rudymatela) [inline tools/send-webhook:139]: Code is misaligned here as well due to spaces instead of tabs:

```suggestion
	[ -n "$state" ] || state=success

	# If this is a URL make sure it has quote marks, if null make sure it doesn't.
	if [ -z "$target" ]; then
		target="null"
	else
		target="\"$target\""
	fi

	event="status"
```

File: tests/Spec.hs
Comment (rudymatela) [inline tests/Spec.hs:1634]: Nice.  The test fixture was already there.  :-)

**Review by rudymatela — COMMENTED** [review]
Nice.  Other than the typo and minor indenting issues I found, looks good to me.  :+1:

File: src/WebInterface.hs
Comment (rudymatela) [reply src/WebInterface.hs:248]: While reading the original issues I found something to improve.

Instead of just "View in CI", maybe it would be better to have: "Build pending" and "Build failed" as the link labels.

**Review by rudymatela — CHANGES_REQUESTED** [review]

**Comment by rudymatela** [comment]
> I'm slightly unsure if this will actually go off in practice. This info is given when the build status changes to pending with a non-null `target_url`, but I have no concrete intuition for when this would occur in practice. I'm hoping when the status check starts it will change the state from `BuildPending Nothing` to `BuildPending (Just url)`.

For this, I think GitHub sends a BuildPending hook again with the build URL when it gets one.  But I think you've already figured this out because I saw your changes on send-webhook which deal with this.

**Review by rudymatela — APPROVED** [review]
LGTM! (modulo minor fixes)

**Comment by rudymatela** [comment]
Btw, I reverted the broken merge of yesterday.  So feel free to rebase and merge this into master now if you like.

I saw this created a few conflicts with your files (as your change was already based on top of mine).  I'm sorry about that :pensive:.  I reviewed the conflicts though, and they seem to be not major.  :-)

**Comment by alex-mckenna** [comment]
Okay, the rebase was mostly mechanical apart from the `handleBuildStatusChanged` which was a little different after the revert for some reason. I changed the PR to be in the style of the reverted version so give it a once over in case anything dumb crept in

**Review by rudymatela — APPROVED** [review]
LGTM.

File: src/Logic.hs
Comment (alex-mckenna) [reply src/Logic.hs:541]: > in case anything dumb crept in

Hey look, something dumb

Comment (rudymatela) [reply src/Logic.hs]: :joy: :sweat_smile:

**Comment by alex-mckenna** [comment]
@OpsBotPrime merge on friday

**Comment by OpsBotPrime** [comment]
Pull request approved for merge by @alex-mckenna, rebasing now.

**Comment by OpsBotPrime** [comment]
Rebased as b4994cabf047f6cdedc1433018f46a2d54d0391b, waiting for CI …

**Pull request merged**
